### ISSUE TITLE
How to use superfluous_disable_command with the analyze command?

### ISSUE THREAD

**Issue by acecilia** [realm/SwiftLint#3765]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I would like to detect when an analyze rule is wrongly disabled. For example, with below code:

```swift
// swiftlint:disable:next unused_import
public struct SomeStruct {
}
```

And below configuration:

```
analyzer_rules:
  - unused_import

only_rules:
  - superfluous_disable_command
```

I would expect swiftlint to raise an error `SwiftLint rule 'unused_import' did not trigger a violation in the disabled region. Please remove the disable command.` on the line where the disable comment is.

But currently no error is raised. Is there a way to do this?

**Comment by jpsim** [comment]
Unfortunately, `superfluous_disable_command` only works with linter rules, not analyzer rules.

**Comment by rgoldberg** [comment]
I think this was implemented (so this issue can be closed), as I recently got:

```text
…: error: Superfluous Disable Command Violation: SwiftLint rule 'unused_declaration' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)
```
